hibernate-core/src/main/java/org/hibernate/sql/ast/tree/expression/ColumnReference.java

```diff
@@ -108,7 +123,7 @@ public boolean equals(Object o) {
 
 	@Override
 	public int hashCode() {
-		return referenceExpression.hashCode();
+		return Objects.hashCode( referenceExpression );
```

Comment:
Revert this. `referenceExpression` is non-null AFAICS

Reply:
I changed this to align well with the `equals()` implementation, which has used `Objects.equals()` already.
I both reverted this and changed the `equals()` implementation to avoid unnecessary nullness checking.